flask/flask-common/src/main/java/net/corda/flask/common/LockFile.java

```diff
@@ -24,21 +25,22 @@ private static RandomAccessFile createLockFile(Path path) throws IOException {
     @SneakyThrows
     public static LockFile acquire(Path path, boolean shared) {
         RandomAccessFile raf = createLockFile(path);
-        return new LockFile(raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
+        return new LockFile(raf, raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
     }
 
     @SneakyThrows
     public static LockFile tryAcquire(Path path, boolean shared) {
         RandomAccessFile raf = createLockFile(path);
         FileLock lock = raf.getChannel().tryLock(0L, Long.MAX_VALUE, shared);
         if(lock != null)
-            return new LockFile(lock);
+            return new LockFile(raf, lock);
         else return null;
     }
 
     @Override
     @SneakyThrows
     public void close() {
         lock.release();
+        raf.close();
```

Comment:
@LWogan has confirmed that
```java
@Override
public void close() {
    lock.channel().close();
}
```
is enough here.

Reply:
yeah, I tested myself on a Windows VM and that works